protostar-rust/cairo-lang-protostar/src/casm_generator/mod.rs

```diff
@@ -0,0 +1,332 @@
+use cairo_lang_sierra::extensions::range_check::RangeCheckType;
+use cairo_lang_sierra::extensions::segment_arena::SegmentArenaType;
+use cairo_lang_sierra::extensions::starknet::syscalls::SystemType;
+use cairo_lang_sierra::extensions::{ConcreteType, NamedType};
+use cairo_lang_sierra::program::Function;
+use cairo_lang_sierra::program_registry::{ProgramRegistry, ProgramRegistryError};
+use cairo_lang_sierra_ap_change::{calc_ap_changes, ApChangeError};
+use cairo_lang_sierra_gas::gas_info::GasInfo;
+use cairo_lang_sierra_to_casm::compiler::{CairoProgram, CompilationError};
+use cairo_lang_sierra_to_casm::metadata::{calc_metadata, Metadata, MetadataError};
+use cairo_lang_utils::bigint::{deserialize_big_uint, serialize_big_uint, BigUintAsHex};
+use cairo_lang_sierra_type_size::{get_type_size_map, TypeSizeMap};
+use cairo_vm::vm::errors::vm_errors::VirtualMachineError;
+use itertools::chain;
+use num_bigint::BigUint;
+use num_integer::Integer;
+use num_traits::{Num, Signed};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum GeneratorError {
+    #[error("Failed calculating gas usage, it is likely a call for `get_gas` is missing.")]
+    FailedGasCalculation,
+    #[error("Function with suffix `{suffix}` to run not found.")]
+    MissingFunction { suffix: String },
+    #[error("Function expects arguments of size {expected} and received {actual} instead.")]
+    ArgumentsSizeMismatch { expected: usize, actual: usize },
+    #[error(transparent)]
+    ProgramRegistryError(#[from] Box<ProgramRegistryError>),
+    #[error(transparent)]
+    SierraCompilationError(#[from] CompilationError),
+    #[error(transparent)]
+    ApChangeError(#[from] ApChangeError),
+    #[error(transparent)]
+    VirtualMachineError(#[from] Box<VirtualMachineError>),
+    #[error("At least one test expected but none detected.")]
+    NoTestsDetected,
+}
+
+#[derive(Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct TestEntrypoint {
+    pub offset: usize,
+    pub name: String,
+}
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct TestConfig {
+    pub name: String,
+    pub available_gas: Option<usize>,
+}
+
+#[derive(Default, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ProtostarCasm {
+    #[serde(serialize_with = "serialize_big_uint", deserialize_with = "deserialize_big_uint")]
+    pub prime: BigUint,
+    pub bytecode: Vec<BigUintAsHex>,
+    pub hints: Vec<(usize, Vec<String>)>,
+    pub test_entry_points: Vec<TestEntrypoint>,
+}
+
+pub struct SierraCasmGenerator {
+    /// The sierra program.
+    sierra_program: cairo_lang_sierra::program::Program,
+    /// Program registry for the Sierra program.
+    sierra_program_registry: ProgramRegistry<CoreType, CoreLibfunc>,
+    /// The casm program matching the Sierra code.
+    casm_program: CairoProgram,
+    /// Program registry for the Sierra program.
+    type_sizes: TypeSizeMap,
+}
```

Comment:
I don't think we use this struct anymore

Reply:
we use it in `fn validate_tests(` and `build_protostar_casm`